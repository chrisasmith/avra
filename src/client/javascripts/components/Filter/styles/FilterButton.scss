@import '../../../../stylesheets/modules/theme.scss';

.button {
  color: #000077;

  &:hover {
    background-color: transparent;
    color: $dark-gray-color;
  }
}