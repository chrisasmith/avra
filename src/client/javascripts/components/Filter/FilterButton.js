import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import { Button } from 'react-bootstrap'
import styles from './styles/FilterButton.scss'

const propTypes = {
  value: PropTypes.number,
  values: PropTypes.array,
  onChange: PropTypes.func.isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  name: PropTypes.string.isRequired,
}

class FilterButton extends Component {
  constructor(props, context) {
    super(props, context)

    this.state = {
      show: true,
    }
  }

  render() {
    return (
      <Button
        className={classnames('btn-outline-primary', styles.button)}
        active={this.props.values.includes(this.props.value)}
        onClick={() => this.props.onChange(this.props.name, this.props.value)}
      >
        {this.props.children}
      </Button>
    )
  }
}

FilterButton.propTypes = propTypes
export default FilterButton
