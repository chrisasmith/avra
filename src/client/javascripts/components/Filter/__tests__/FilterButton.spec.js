import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import FilterButton from '../FilterButton'

describe('Components FilterButton', () => {
  it('should render', () => {
    const wrapper = shallow(<FilterButton
      value={1}
      values={[1, 2]}
      onChange={jest.fn()}
      name="thingy"
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('fire change', () => {
    const onChange = jest.fn()
    const wrapper = mount(<FilterButton
      value={1}
      values={[1, 2]}
      onChange={onChange}
      name="thingy"
    />)

    wrapper.find('Button').simulate('click')
    expect(onChange).toHaveBeenCalledWith('thingy', 1)
  })

  it('should render inactive', () => {
    const wrapper = shallow(<FilterButton
      value={1}
      values={[2]}
      onChange={jest.fn()}
      name="thingy"
    />)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should render children', () => {
    const wrapper = shallow(<FilterButton
      value={1}
      values={[2]}
      onChange={jest.fn()}
      name="thingy"
    ><div>hi</div></FilterButton>)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
