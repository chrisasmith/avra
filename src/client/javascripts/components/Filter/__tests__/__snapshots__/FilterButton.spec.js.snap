// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components FilterButton should render 1`] = `
<ForwardRef(Bootstrap(Button))
  active={true}
  className="btn-outline-primary button"
  onClick={[Function]}
/>
`;

exports[`Components FilterButton should render children 1`] = `
<ForwardRef(Bootstrap(Button))
  active={false}
  className="btn-outline-primary button"
  onClick={[Function]}
>
  <div>
    hi
  </div>
</ForwardRef(Bootstrap(Button))>
`;

exports[`Components FilterButton should render inactive 1`] = `
<ForwardRef(Bootstrap(Button))
  active={false}
  className="btn-outline-primary button"
  onClick={[Function]}
/>
`;
